### PR TITLE
added back the code

### PR DESCRIPTION
**Pull request by edwinclement08**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const { port, hostname } = new URL(`http://${req.url}`);
+
+  dns.lookup(hostname, (err, address) => {
+      // needed as Pi Hole converts all black listed ips to localhost
+    if (( !err && address !== '0.0.0.0') && hostname !== '0.0.0.0') {
+        const serverSocket = net.connect(port || 80, hostname, () => {
+            clientSocket.write(
+            "HTTP/1.1 200 Connection Established\r\n" +
+                "Proxy-agent: Node.js-Proxy\r\n" +
+                "\r\n"
```
Comment (Raj-7799) [inline index.js:48]: Didn't understand the actual use-case of performing dns check here

Comment (Raj-7799) [reply index.js:48]: Is it to safeguard proxying to itself?

Comment (edwinclement08) [reply index.js:48]: ah no
i have pi hole
that redirects blacklisted domains to 0.0.0.0
which crashes the server as there is no server at 0.0.0.0:80 or :443

**Review by Raj-7799 — CHANGES_REQUESTED** [review]
Performing curl intermittently fails `curl -x 127.0.0.1:3000 https://www.google.com`
```
❯ node index.js
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:111:27)
Emitted 'error' event at:
    at errorOrDestroy (internal/streams/destroy.js:107:12)
    at Socket.onerror (_stream_readable.js:717:7)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

File: index.js (hunk 1 above)
Comment (Raj-7799) [reply index.js:48]: Got it.

**Review by Raj-7799 — APPROVED** [review]
lgtm! Good backpressure implementation

**Pull request merged**